src/main.c

```diff
@@ -1231,6 +1238,9 @@ int main(int argc, char * argv [])
       free(port_tok[i]);
   }
 
+  if(touch_1200bps && pgm->conntype == CONNTYPE_SERIAL)
```

Comment:
Maybe move into the preceding if(pgm->conntype == CONNTYPE_SERIAL) block? Simplifies code
```
if(touch_1200bps)
  touch_serialport(&port, 1200);
```
(and saves an empty line)

Reply:
And, of course, we should resolve whether a failed touch is 
 - Fatal: use `pmsg_error()` in `touch_serialport()` and exit:
   ```
   if(touch_1200bps && touch_serialport(&port, 1200) != 0)
      exit(1);
   ```
 - Advisory only: ignore failure but then use `pmsg_warning()` in `touch_serialport()`

I lean towards the fist as `-r` was requested and could not be delivered.